apollo-router/src/plugins/telemetry/config_new/attributes.rs

```diff
@@ -0,0 +1,702 @@
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    ResponseBody {
+        /// Json Path into the response body
+        response_body: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    RequestHeader {
+        /// The name of the request header.
+        request_header: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    ResponseHeader {
+        /// The name of the response header.
+        response_header: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    RequestContext {
+        /// The request context key.
+        request_context: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<AttributeValue>,
+    },
+    ResponseContext {
+        /// The response context key.
+        response_context: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<AttributeValue>,
+    },
+    Baggage {
+        /// The name of the baggage item.
+        baggage: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    Env {
+        /// The name of the environment variable
+        env: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, JsonSchema, Clone, Debug)]
+#[serde(deny_unknown_fields, rename_all = "snake_case", untagged)]
+pub(crate) enum SubgraphCustomAttribute {
+    SubgraphOperationName {
+        /// The operation name from the subgraph query.
+        subgraph_operation_name: OperationName,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    SubgraphOperationKind {
+        /// The kind of the subgraph operation (query|mutation|subscription).
+        subgraph_operation_kind: OperationKind,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    SubgraphQueryVariable {
+        /// The name of a subgraph query variable.
+        subgraph_query_variable: String,
+        /// Optional redaction pattern.
+        redact: Option<String>,
+        /// Optional default value.
+        default: Option<String>,
+    },
+    SubgraphResponseBody {
+        /// The subgraph response body json path.
+        subgraph_response_body: String,
+        /// Optional default value.
```

Comment:
Should we also have a way to `redact` it ?

Reply:
We have a `redact` for the supergraph version

Reply:
[210cac9](https://github.com/apollographql/router/pull/4061/commits/210cac92202f7130b601728e934bab11c15fc74d)